airbyte-integrations/connectors/source-bigquery/source_bigquery/source.py

```diff
@@ -82,31 +113,105 @@ def check_connection(self, logger, config: Mapping[str, Any]) -> Tuple[bool, any
             return True, error_msg
         return True, None
     
-    def streams(self, config: Mapping[str, Any]) -> Iterable[Stream]:
+    def streams(self, config: Mapping[str, Any]) -> List[Stream]:
         """
         Replace the streams below with your own streams.
 
         :param config: A Mapping of the user input configuration as defined in the connector spec.
         """
+        project_id = config["project_id"]
         self._auth = BigqueryAuth(config)
         streams = config.get("streams", [])
         sync_method = config["replication_method"]["method"]
-        streams_catalog = []
-        
-        for dataset in BigqueryDatasets(project_id=config["project_id"], authenticator=self._auth).read_records(sync_mode=SyncMode.full_refresh):
+        slice_range = float(config.get("slice_range", 525600)) # 1 year by default
+        fallback_start =  datetime.strptime("0001-01-01T00:00:00.000Z", '%Y-%m-%dT%H:%M:%S.%f%z')
+        change_history_start = datetime.now(tz=pytz.timezone("UTC")) - timedelta(days=7) + timedelta(seconds=30)
+        normal_streams = []
+        concurrent_streams = []
+
+        for stream in streams:
+                dataset_id, table_id = stream['parent_stream'].split(".")
+                where_clause = stream["where_clause"]
+                stream_name = stream["name"]
+                if sync_method == "Standard":
+                    table_obj = IncrementalQueryResult(project_id, dataset_id, table_id, stream_name, where_clause, fallback_start=fallback_start, slice_range=slice_range, authenticator=self._auth)
+                else:
+                    try:
+                        table_obj = TableChangeHistory(project_id, dataset_id, table_id, stream_name, where_clause=where_clause, fallback_start=change_history_start, slice_range=slice_range, authenticator=self._auth)
+                        next(table_obj.read_records(sync_mode=SyncMode.full_refresh))
+                    except exceptions.HTTPError as error:
+                        self.logger.warning(f"Table had no inserts in the last seven days so no change history is available causing the error: {str(error.response.text)}")
```

Comment:
The error message makes me worried a bit. It is not because there are no updated in the last seven days that we should take action. One issue would be if we were to query before the retention date because the sync frequency is too infrequent. 

Can you clarify which case was problematic with not having data in the past seven days? Can we add a mock server test to document the conditions and behavior we have based on those conditions?

Reply:
@maxi297
I think it's better to remove the warning in this case and only do it when we query before the retention date which is the problematic case as you said. Otherwise we return nothing if there are no changes during the time travel window if we have no state